ssg/templates.py

```diff
@@ -12,22 +12,23 @@
 from ssg.build_cpe import ProductCPEs
 from collections import namedtuple
 
-templating_lang = namedtuple(
+
+TemplatingLang = namedtuple(
     "templating_language_attributes",
     ["name", "file_extension", "template_type", "lang_specific_dir"])
 
-template_type = ssg.utils.enum("remediation", "check")
-
-languages = {
-    "anaconda": templating_lang("anaconda", ".anaconda", template_type.remediation, "anaconda"),
-    "ansible": templating_lang("ansible", ".yml", template_type.remediation, "ansible"),
-    "bash": templating_lang("bash", ".sh", template_type.remediation, "bash"),
-    "blueprint": templating_lang("blueprint", ".toml", template_type.remediation, "blueprint"),
-    "ignition": templating_lang("ignition", ".yml", template_type.remediation, "ignition"),
-    "kubernetes": templating_lang("kubernetes", ".yml", template_type.remediation, "kubernetes"),
-    "oval": templating_lang("oval", ".xml", template_type.check, "oval"),
-    "puppet": templating_lang("puppet", ".pp", template_type.remediation, "puppet"),
-    "sce-bash": templating_lang("sce-bash", ".sh", template_type.remediation, "sce")
+TemplateType = ssg.utils.enum("REMEDIATION", "CHECK")
+
+LANGUAGES = {
+    "anaconda": TemplatingLang("anaconda", ".anaconda", TemplateType.REMEDIATION, "anaconda"),
+    "ansible": TemplatingLang("ansible", ".yml",        TemplateType.REMEDIATION, "ansible"),
+    "bash": TemplatingLang("bash", ".sh",               TemplateType.REMEDIATION, "bash"),
+    "blueprint": TemplatingLang("blueprint", ".toml",   TemplateType.REMEDIATION, "blueprint"),
+    "ignition": TemplatingLang("ignition", ".yml",      TemplateType.REMEDIATION, "ignition"),
+    "kubernetes": TemplatingLang("kubernetes", ".yml",  TemplateType.REMEDIATION, "kubernetes"),
+    "oval": TemplatingLang("oval", ".xml",              TemplateType.CHECK,       "oval"),
+    "puppet": TemplatingLang("puppet", ".pp",           TemplateType.REMEDIATION, "puppet"),
+    "sce-bash": TemplatingLang("sce-bash", ".sh",       TemplateType.REMEDIATION, "sce")
```

Comment:
Not suggesting any change, as it is code from before, but isn't it strange that "script check engine" is treated as "remediation"? The template type probably means something else than what it says.

Reply:
Good point. No idea, but it might be somehow connected to the way we process shell scripts. SCE implementation has a lot of quirks all over the place.

Reply:
It might be a bug, it's used in _init_lang_output_dirs in the following way:

```
            if lang.template_type == TemplateType.CHECK:
                output_dir = self.checks_dir
            else:
                output_dir = self.remediations_dir
```

It might be hidden because we don't have any SCE template in our repository.

Reply:
But let's not change it in this PR.